base/freestanding/femr/config/fhir/application.yaml

```diff
@@ -1,84 +1,128 @@
+  #      hibernate.cache.use_minimal_puts: false
+  ###    These settings will enable fulltext search with lucene or elastic
+      hibernate.search.enabled: true
+  ### lucene parameters
 #      hibernate.search.backend.type: lucene
-#      hibernate.search.backend.analysis.configurer: ca.uhn.fhir.jpa.search.HapiLuceneAnalysisConfigurer
+#      hibernate.search.backend.analysis.configurer: ca.uhn.fhir.jpa.search.HapiHSearchAnalysisConfigurers$HapiLuceneAnalysisConfigurer
 #      hibernate.search.backend.directory.type: local-filesystem
 #      hibernate.search.backend.directory.root: target/lucenefiles
 #      hibernate.search.backend.lucene_version: lucene_current
-  batch:
-    job:
-      enabled: false
-  main:
-#    TODO 5.6.0 -> Prevent duplicate bean definitions in the Spring batch config in HAPI: see:
-    allow-bean-definition-overriding: true
+  ### elastic parameters ===> see also elasticsearch section below <===
+#      hibernate.search.backend.type: elasticsearch
+#      hibernate.search.backend.analysis.configurer: ca.uhn.fhir.jpa.search.HapiHSearchAnalysisConfigurers$HapiElasticAnalysisConfigurer
 hapi:
   fhir:
     ### This enables the swagger-ui at /fhir/swagger-ui/index.html as well as the /fhir/api-docs (see https://hapifhir.io/hapi-fhir/docs/server_plain/openapi.html)
     openapi_enabled: true
     ### This is the FHIR version. Choose between, DSTU2, DSTU3, R4 or R5
     fhir_version: R4
-### enable to use the ApacheProxyAddressStrategy which uses X-Forwarded-* headers
-### to determine the FHIR server address
-#   use_apache_address_strategy: false
-### forces the use of the https:// protocol for the returned server address.
-### alternatively, it may be set using the X-Forwarded-Proto header.
-#   use_apache_address_strategy_https: false
-### enable to set the Server URL
-#    server_address: http://hapi.fhir.org/baseR4
-#    defer_indexing_for_codesystems_of_size: 101
-#    install_transitive_ig_dependencies: true
-#    implementationguides:
-###    example from registry (packages.fhir.org)
-#      swiss:
-#        name: swiss.mednet.fhir
-#        version: 0.8.0
-#      example not from registry
-#      ips_1_0_0:
-#        url: https://build.fhir.org/ig/HL7/fhir-ips/package.tgz
-#        name: hl7.fhir.uv.ips
-#        version: 1.0.0
-#    supported_resource_types:
-#      - Patient
-#      - Observation
-#    allow_cascading_deletes: true
-#    allow_contains_searches: true
-#    allow_external_references: true
-#    allow_multiple_delete: true
-#    allow_override_default_search_params: true
-#    auto_create_placeholder_reference_targets: false
-#    cql_enabled: true
-#    default_encoding: JSON
-#    default_pretty_print: true
-#    default_page_size: 20
-#    delete_expunge_enabled: true
-#    enable_repository_validating_interceptor: false
-#    enable_index_missing_fields: false
-#    enable_index_contained_resource: false
-    enforce_referential_integrity_on_delete: false
```

Comment:
i don't love that we had `enforce_referential_integrity_on_delete` disabled before, but it does look like that setting changed with this PR, fwiw.

Reply:
Thanks! We do need to keep that until the FHIR REST API allows cascading deletes

I switched that config to be set by an env var (vs config file) and added a TODO, so we don't have to worry about overwriting it now